baseplate/clients/requests.py

```diff
@@ -0,0 +1,322 @@
+
+        """
+        return self.request("OPTIONS", url, **kwargs)
+
+    def patch(self, url: str, **kwargs: Any) -> Response:
+        """Send a PATCH request.
+
+        See :py:func:`requests.request` for valid keyword arguments.
+
+        """
+        return self.request("PATCH", url, **kwargs)
+
+    def put(self, url: str, **kwargs: Any) -> Response:
+        """Send a PUT request.
+
+        See :py:func:`requests.request` for valid keyword arguments.
+
+        """
+        return self.request("PUT", url, **kwargs)
+
+    def prepare_request(self, request: Request) -> PreparedRequest:
+        """Construct a :py:class:`~requests.PreparedRequest` for later use.
+
+        The prepared request can be stored or manipulated and then used with
+        :py:meth:`send`.
+
+        """
+        return request.prepare()
+
+    def request(self, method: str, url: Union[str, bytes], **kwargs: Any) -> Response:
+        """Send a request.
+
+        :param method: The HTTP method of the request, e.g. ``GET``, ``PUT``, etc.
+        :param url: The URL to send the request to.
+
+        See :py:func:`requests.request` for valid keyword arguments.
+
+        """
+        request = Request(method=method.upper(), url=url, **kwargs)
+        prepared = self.prepare_request(request)
+        return self.send(prepared)
+
+    def send(self, request: PreparedRequest) -> Response:
+        """Send a :py:class:`~requests.PreparedRequest`."""
+
+        with self.span.make_child(f"{self.name}.request") as span:
+            span.set_tag("http.method", request.method)
+            span.set_tag("http.url", request.url)
+
+            # we cannot re-use the same session every time because sessions re-use the same
+            # CookieJar and so we'd muddle cookies cross-request. if the application wants
+            # to keep track of cookies, it should do so itself.
+            #
+            # note: we're still getting connection pooling because we're re-using the adapter.
+            session = Session()
+            session.mount("http://", self.adapter)
+            session.mount("https://", self.adapter)
+            response = session.send(request)
+
+            span.set_tag("http.status_code", response.status_code)
+        return response
+
+
+class InternalBaseplateSession(BaseplateSession):
+    def send(self, request: PreparedRequest) -> Response:
+        request.headers["X-Trace"] = str(self.span.trace_id)
+        request.headers["X-Parent"] = str(self.span.parent_id)
+        request.headers["X-Span"] = str(self.span.id)
+        if self.span.sampled:
+            request.headers["X-Sampled"] = "1"
+        if self.span.flags is not None:
+            request.headers["X-Flags"] = str(self.span.flags)
+
+        try:
+            edge_context = self.span.context.raw_request_context
+        except AttributeError:
+            pass
+        else:
+            request.headers["X-Edge-Context"] = base64.b64encode(edge_context).decode()
```

Comment:
Should this be `X-Edge-Request` to be consistent with what's in pyramid.py?

Reply:
well, that's super embarrassing. you're absolutely right.

i've updated this and also reworked the test suite a bit to use baseplate-integrated pyramid on the receiving end with higher-level tests of correctness. this made it obvious the headers we were sending were also wrong: they were for the parent span not the child span. so i've fixed that as well.

thanks for calling this out!